lib/TWCManager/TWCMaster.py

```diff
@@ -157,6 +157,30 @@ def checkScheduledCharging(self):
                     blnUseScheduledAmps = 1
         return blnUseScheduledAmps
 
+    def checkVINEntitlement(self, subTWC):
+        # When provided with the TWC that has had the VIN reported for a vehicle
+        # we check the policy for charging and determine if it is allowed or not
+
+        if not subTWC.currentVIN:
+            # No VIN supplied. We can't make any decision other than allow
```

Comment:
Just to be clear, this is distinct from the all-zero VIN you get on a non-Tesla charging, correct?

Reply:
This is a great point @MikeBishop. While it would not match this case (a non Tesla vehicle returns the string of zeros as you mention - which evaluates to True vs a charger without VIN functionality not responding at all and the empty string evaluates to False), I think I should add a dedicated setting / special case for non Tesla vehicles, either by creating a dummy vehicle with all zeros VINs to track non Tesla vehicles and their charging rate, or a checkbox that allows/denies charging to non Tesla vehicles